src/query.c

```diff
@@ -1539,10 +1539,15 @@ void QueryNode_SetFieldMask(QueryNode *n, t_fieldMask mask) {
 void QueryNode_AddChildren(QueryNode *n, QueryNode **children, size_t nchildren) {
   if (n->type == QN_TAG) {
     for (size_t ii = 0; ii < nchildren; ++ii) {
-      if (children[ii]->type == QN_TOKEN || children[ii]->type == QN_PHRASE ||
-          children[ii]->type == QN_PREFIX || children[ii]->type == QN_LEXRANGE ||
-          children[ii]->type == QN_WILDCARD_QUERY) {
+      QueryNode *child = children[ii];
+      if (child->type == QN_TOKEN || child->type == QN_PHRASE ||
+          child->type == QN_PREFIX || child->type == QN_LEXRANGE ||
+          child->type == QN_WILDCARD_QUERY) {
         n->children = array_ensure_append(n->children, children + ii, 1, QueryNode *);
+        for(size_t jj = 0; jj < QueryNode_NumParams(child); ++jj) {
+          Param *p = child->params + jj;
+          p->type = PARAM_TERM_CASE;
```

Comment:
When evaluating `PARAM_TERM_CASE`  we use `rm_strdup` and not `rm_strndup_unescape` 
https://github.com/RediSearch/RediSearch/blob/7eae2f60a71bf5122e369faa3ce0e8baa5a94e6c/src/query_param.c#L176
So we do not unescape params?